File: Backend - DotNet API/EmployeeDirectory.Models/Employee.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace EmployeeDirectory.Models
{
    public partial class Employee
    {
        public int Id { get; set; }
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public int JobTitleId { get; set; }
        public int OfficeLocationId { get; set; }
        public int DepartmentId { get; set; }
        public long PhoneNumber { get; set; }
        public string SkypeId { get; set; } = string.Empty;

        // Navigation property
        [ForeignKey("DepartmentId")]
        public Department Department { get; set; }
        
        [ForeignKey("JobTitleId")]
        public JobTitle JobTitle { get; set; }

        [ForeignKey("OfficeLocationId")]
        public OfficeLocation OfficeLocation { get; set; }
    }
}

File: Frontend - Angular UI/src/app/model/employee.ts
export class Employee {
    id: number;
    firstName: string;
    lastName: number;
    email: string;
    jobTitleId: number
    officeLocationId: number
    departmentId: number
    phoneNumber: any
    skypeId: string

    constructor(args: any) {
        this.id = args.Id;
        this.firstName = args.firstName;
        this.lastName = args.lastName;
        this.email = args.email
        this.jobTitleId = args.jobTitleId
        this.officeLocationId = args.officeLocationId
        this.departmentId = args.departmentId
        this.phoneNumber = args.phoneNumber
        this.skypeId = args.skypeId
    }
}
